Penumbra/Import/Textures/CombinedTexture.Manipulation.cs

```diff
@@ -206,26 +243,59 @@ private void MultiplyPixelsRight(int y, ParallelLoopState _)
         }
     }
 
+    private byte[] ResizePixels(byte[] rgbaPixels, int sourceWidth, int sourceHeight)
+    {
+        if (sourceWidth == _targetWidth && sourceHeight == _targetHeight)
+            return rgbaPixels;
+
+        byte[] resizedPixels;
+        using (var image = Image.LoadPixelData<Rgba32>(rgbaPixels, sourceWidth, sourceHeight))
+        {
+            image.Mutate(ctx => ctx.Resize(_targetWidth, _targetHeight));
+
+            resizedPixels = new byte[_targetWidth * _targetHeight * 4];
+            image.CopyPixelDataTo(resizedPixels);
+        }
+
+        return resizedPixels;
+    }
+
 
     private (int Width, int Height) CombineImage()
     {
         var combineOp = GetActualCombineOp();
-        var (width, height) = combineOp is not CombineOp.Invalid or CombineOp.RightCopy or CombineOp.RightMultiply
+        var (usesLeft, usesRight) = GetCombineOpFlags(combineOp);
+        var resizeOp = usesLeft && usesRight ? _resizeOp : ResizeOp.None;
+        (_targetWidth, _targetHeight) = usesLeft && resizeOp != ResizeOp.ToRight
             ? (_left.TextureWrap!.Width, _left.TextureWrap!.Height)
             : (_right.TextureWrap!.Width, _right.TextureWrap!.Height);
-        _centerStorage.RgbaPixels = new byte[width * height * 4];
+        _centerStorage.RgbaPixels = new byte[_targetWidth * _targetHeight * 4];
         _centerStorage.Type       = TextureType.Bitmap;
-        Parallel.For(0, height, combineOp switch
+        try
+        {
+            if (usesLeft)
```

Comment:
Is all this necessary? ResizePixels does not do anything if the sizes are unchanged, so just use the _targetWidth and _targetHeight here and let the function handle that, we don't need to check the resizeop anymore, do we? Same for right?

(I think I misclicked here and started this as a review instead of a single comment, but did not publish the review or something?)

Reply:
It is unneeded indeed, as `ToLeft` means the target size is the same as the left size, which makes `ResizePixels` a no-op, and ditto with `None when usesLeft`. Good catch, though the downside is that it makes the intent less explicit.